radixdlt-core/radixdlt/src/main/java/com/radixdlt/environment/Dispatchers.java

```diff
@@ -79,14 +80,26 @@ private Dispatchers() {
 		@Inject
 		private Provider<Environment> environmentProvider;
 		private final Class<T> c;
+		private final TypeLiteral<T> t;
 
 		ScheduledDispatcherProvider(Class<T> c) {
 			this.c = c;
+			this.t = null;
+		}
+
+		ScheduledDispatcherProvider(TypeLiteral<T> t) {
+			this.c = null;
+			this.t = t;
 		}
 
 		@Override
 		public ScheduledEventDispatcher<T> get() {
-			return environmentProvider.get().getScheduledDispatcher(c);
+			Environment e = environmentProvider.get();
+			if (c != null) {
+				return e.getScheduledDispatcher(c);
+			} else {
+				return e.getScheduledDispatcher(t);
+			}
```

Comment:
I think `c` and `t` should have more informative names, as catching small (one letter) difference between return statements is rather inconvenient. Also, this should help to realize that actually here can be invoked two different methods rather than one with different parameters.